src/VisualStudio/CSharp/Test/UnifiedSettings/CSharpUnifiedSettingsTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Editor.CSharp.CompleteStatement;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.UnitTests.UnifiedSettings;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Roslyn.VisualStudio.CSharp.UnitTests.UnifiedSettings
+{
+    public class CSharpUnifiedSettingsTests : UnifiedSettingsTests
+    {
+        internal override ImmutableArray<IOption2> OnboardedOptions => ImmutableArray.Create<IOption2>(
+            CompletionOptionsStorage.TriggerOnTypingLetters,
+            CompletionOptionsStorage.TriggerOnDeletion,
+            CompletionOptionsStorage.TriggerInArgumentLists,
+            CompletionViewOptionsStorage.HighlightMatchingPortionsOfCompletionListItems,
+            CompletionViewOptionsStorage.ShowCompletionItemFilters,
+            CompleteStatementOptionsStorage.AutomaticallyCompleteStatementOnSemicolon,
+            CompletionOptionsStorage.SnippetsBehavior,
+            CompletionOptionsStorage.EnterKeyBehavior,
+            CompletionOptionsStorage.ShowNameSuggestions,
+            CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces,
+            CompletionViewOptionsStorage.EnableArgumentCompletionSnippets,
+            CompletionOptionsStorage.ShowNewSnippetExperienceUserOption
+        );
+
+        internal override ImmutableDictionary<IOption2, object> OptionsToDefaultValue => ImmutableDictionary<IOption2, object>.Empty.
+            Add(CompletionOptionsStorage.SnippetsBehavior, SnippetsRule.AlwaysInclude).
+            Add(CompletionOptionsStorage.EnterKeyBehavior, EnterKeyRule.Never).
+            Add(CompletionOptionsStorage.TriggerOnDeletion, false).
+            Add(CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces, true).
+            Add(CompletionViewOptionsStorage.EnableArgumentCompletionSnippets, false).
+            Add(CompletionOptionsStorage.ShowNewSnippetExperienceUserOption, false);
```

Comment:
Why include these? The default values are defined as part of the option, so it doesn't seem necessary to duplicate that.

Reply:
It's not always true. In many cases we override the default value at runtime, but in unified settings the default value are always put in the registration json.
Case 1:
`CompletionOptionsStorage.TriggerOnDeletion`  Its default value is null, the real default values are set different for [VB](https://github.com/dotnet/roslyn/blob/efdb56597abbe7dff8f5df311f60375877f0b17d/src/VisualStudio/VisualBasic/Impl/Options/IntelliSenseOptionPageControl.xaml.vb#L18) and [C#](https://github.com/dotnet/roslyn/blob/efdb56597abbe7dff8f5df311f60375877f0b17d/src/VisualStudio/CSharp/Impl/Options/IntelliSenseOptionPageControl.xaml.cs#L23). 
The gap here I think is `PerLanguageOption` can only have one default value, but in reality sometimes we needs different default value for each lang.

Case 2: 
`CompletionOptionsStorage.ShowNewSnippetExperienceUserOption` Its default value is null, we use the feature flag trick, the real effect value if `false

Reply:
Done